src/commands/sites/sites-create-template.js

```diff
@@ -0,0 +1,251 @@
+    let siteSuggestion
+    if (!user) user = await api.getCurrentUser()
+
+    if (!name) {
+      let { slug } = user
+      let suffix = ''
+
+      // If the user doesn't have a slug, we'll compute one. Because `full_name` is not guaranteed to be unique, we
+      // append a short randomly-generated ID to reduce the likelihood of a conflict.
+      if (!slug) {
+        slug = slugify(user.full_name || user.email)
+        suffix = `-${uuidv4().slice(0, SITE_NAME_SUGGESTION_SUFFIX_LENGTH)}`
+      }
+
+      const suggestions = [
+        `super-cool-site-by-${slug}${suffix}`,
+        `the-awesome-${slug}-site${suffix}`,
+        `${slug}-makes-great-sites${suffix}`,
+        `netlify-thinks-${slug}-is-great${suffix}`,
+        `the-great-${slug}-site${suffix}`,
+        `isnt-${slug}-awesome${suffix}`,
+      ]
+      siteSuggestion = sample(suggestions)
+
+      console.log(
+        `Choose a unique site name (e.g. ${siteSuggestion}.netlify.app) or leave it blank for a random name. You can update the site name later.`,
+      )
+      const { name: nameInput } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'name',
+          message: 'Site name (optional):',
+          filter: (val) => (val === '' ? undefined : val),
+          validate: (input) => /^[a-zA-Z\d-]+$/.test(input) || 'Only alphanumeric characters and hyphens are allowed',
+        },
+      ])
+      name = nameInput
+    }
+
+    try {
+      const siteName = name ? name.trim() : siteSuggestion
+
+      // Create new repo from template
+      const createGhRepoResp = await fetch(`https://api.github.com/repos/${templateUrl.templateName}/generate`, {
+        method: 'POST',
+        headers: {
+          Authorization: `token ${ghToken.token}`,
+        },
+        body: JSON.stringify({
+          name: siteName,
+        }),
+      })
+      const resp = await createGhRepoResp.json()
+
+      if (resp.errors) {
+        if (resp.errors[0].includes('Name already exists on this account')) {
+          throw new Error('Duplicate repo')
+        }
+      } else {
+        site = await api.createSiteInTeam({
+          accountSlug,
+          body: {
+            repo: {
+              provider: 'github',
+              repo: resp.full_name,
+              private: resp.private,
+              branch: resp.default_branch,
+            },
+            name: siteName,
+          },
+        })
+      }
+    } catch (error_) {
+      if (error_.status === 422 || error_.message === 'Duplicate repo') {
+        warn(
+          `${name}.netlify.app already exists or a repository named ${name} already exists on this account. Please try a different slug.`,
+        )
+        await inputSiteName()
+      } else {
+        error(`createSiteInTeam error: ${error_.status}: ${error_.message}`)
+      }
+    }
+  }
+  // In case the user picked a site's name that already existed, we shouldn't ask to authenticate again.
+  if (!ghToken) {
```

Comment:
For some reason, I thought that if you authenticated a first time, it would somehow remember for other commands you'd wanna run after but you're right, it doesn't!